src/lib/select/select.ts

```diff
@@ -758,6 +782,45 @@ export class MatSelect extends _MatSelectMixinBase implements AfterContentInit,
     return !this._selectionModel || this._selectionModel.isEmpty();
   }
 
+  /** Show the backdrop of the overlay. */
+  private _showBackdrop() {
```

Comment:
Not necessarily something for this PR, but rather than adding something special for the select, we can add the ability to detach and re-attach the backdrop via the `OverlayRef`.

Reply:
I tried detach and re-attach the backdrop via the `OverlayRef` and I think there's a risk of detaching other overlay's backdrop accidentally or failed to detach correctly since there're some async activities outside angular zone. It's more reliable to make the backdrop visible/invisible.

Reply:
If there's a risk that that detaching one overlay could unintentionally detach another, that sounds like a larger problem with the cdk or its implementation here.

Reply:
I'm not sure I see how that could happen; each OverlayRef has a reference to its own specific backdrop element

Reply:
Each OverlayRef has its own backdrop. The detach backdrop is an async call 
`this._ngZone.runOutsideAngular(() => setTimeout(finishDetach, 500));` 
When user click on the backdrop, it takes 500ms to fully detach the overlay. If in that 500ms user open the select panel again, a backdrop stilll exists so we cannot reattach it, and is removed after 500ms which means click outside the select panel will not close the panel.

Reply:
Do we know why the CDK waits 500ms to call `finishDetach` when there's no transition? I would think it should finish detaching immediately if there's no transition.

Reply:
There's transition. 

`backdropToDetach.classList.remove('cdk-overlay-backdrop-showing');` This triggers transition and `finishDetach` is triggered when the `transitionend`. 
https://github.com/angular/material2/blob/master/src/cdk/overlay/overlay-ref.ts#L348

Reply:
I see. So would you say the deeper issue here is that `this._ngZone.runOutsideAngular(() => setTimeout(finishDetach, 500));` is meant to accommodate situations where there is no `transitionend` but currently it runs regardless of whether there's a transition or not?

Reply:
What we were trying to avoid with that `setTimeout` is a call to `getComputedStyle` just so we can figure out whether the backdrop has a transition. I had a PR a while ago that switched the backdrop to be a component which would've given us better control over the animation, however we never managed to get it presubmitted (see https://github.com/angular/material2/pull/6627).

Reply:
@crisbeto Do you think it would work better if, instead of having a 500ms timeout, there was a 10ms timeout that gets cleared with `clearTimeout()` if `transitionstart` is fired?

Reply:
If I remember correctly, `transitionstart` was flaky between browsers which means that if we started depending on it, people's tests would start flaking as well.